src/explore-education-statistics-common/src/components/Details.tsx

```diff
@@ -71,8 +64,14 @@ const Details = ({
   }, []);
 
   useEffect(() => {
-    setOpened(open);
-  }, [open, setOpened]);
+    toggleOpen(defaultOpen);
+  }, [defaultOpen, toggleOpen]);
+
+  useEffect(() => {
+    if (preventToggle) {
+      toggleOpen(true);
+    }
+  }, [preventToggle, toggleOpen]);
```

Comment:
Would merge these into a single effect. If the parent component changes the `preventToggle` and `open` props, these would probably interfere with one another.